generative_ai/embedding.py

```diff
@@ -13,18 +13,28 @@
 # limitations under the License.
 
 # [START aiplatform_sdk_embedding]
-from vertexai.language_models import TextEmbeddingModel
+from typing import List
 
-
-def text_embedding(text: str = "What is life?") -> list:
-    """Text embedding with a Large Language Model."""
-    model = TextEmbeddingModel.from_pretrained("textembedding-gecko@001")
-    embeddings = model.get_embeddings([text])
-    vector = embeddings[0].values
-    print(f"Length of Embedding Vector: {len(vector)}")
-    return vector
+from vertexai.language_models import TextEmbeddingInput, TextEmbeddingModel
 
 
+def embed_text(
+    texts: List[str] = ["banana muffins? ", "banana bread? banana muffins?"],
```

Comment:
I'm not sure it makes sense to keep these as separate lists, even though they are zipped together later, I would prefer these being more clear of which task type goes with which text.

And I would also recommend keeping it to a single example instead of multiple, it's easier to read/follow.

Reply:
Done. Revised to process a task type for a batch prediction of a number of texts.